renderdoc/driver/d3d12/d3d12_pixelhistory.cpp

```diff
@@ -573,25 +573,50 @@ struct D3D12PixelHistoryCallback : public D3D12ActionCallback
       baseSlice = 0;
     }
 
-    // For pipeline barriers.
-    D3D12_RESOURCE_BARRIER barrier = {};
-    barrier.Type = D3D12_RESOURCE_BARRIER_TYPE_TRANSITION;
-    barrier.Transition.pResource = p.srcImage;
-    barrier.Transition.StateBefore = p.srcImageState;
-    barrier.Transition.StateAfter = D3D12_RESOURCE_STATE_COPY_SOURCE;
-    barrier.Transition.Subresource =
-        D3D12CalcSubresource(baseMip, baseSlice, p.planeSlice, m_CallbackInfo.targetDesc.MipLevels,
-                             m_CallbackInfo.targetDesc.DepthOrArraySize);
-
     // Multi-sampled images can't call CopyTextureRegion for a single sample, so instead
     // copy using a compute shader into a staging image first
     if(p.multisampled)
     {
-      // TODO: Is a resource transition needed here?
+      // For pipeline barriers.
+      D3D12_RESOURCE_BARRIER barriers[2];
```

Comment:
Minor thing: RenderDoc style wise it is common to do this as
`D3D12_RESOURCE_BARRIER barriers[2] = {};`
which should zero initialise all members.

Reply:
Will do.